MovingBlocks/Terasology
# include dependency versions when defining a module's gradle dependencies

When building a module, gradle was using the module's module.txt to get the _names_ of the module's dependencies, but it was ignoring the versions there.

This PR adds those version constraints to gradle's dependency information.


### How to test

Build modules from source!

Build modules from source that depend on other source modules in your workspace!

Throw some garbage (like git conflict markers) in your `modules.txt` file, make sure the error message is not misleading!

Build modules from source that depend on modules _not_ in your workspace to make it get their jar files from the repo!

Run your module tests!


I think this is pretty likely to break some builds. The question is, will those breakages turn out to reveal bad dependency data? or will those be signs of bugs in this PR?


### Outstanding

- [x] check error handling on malformed module.txt
- [ ] make sure modules with `isReleaseManaged: true` are handled appropriately, e.g. [CoreSampleGameplay](https://github.com/Terasology/CoreSampleGameplay/blob/develop/module.txt)
